build/azure-pipelines/product-compile.yml

```diff
@@ -23,7 +23,7 @@ steps:
     condition: and(succeeded(), ne(variables['NPM_REGISTRY'], 'none'))
     displayName: Setup NPM Registry
 
-  - script: mkdir -p .build && node build/azure-pipelines/common/computeNodeModulesCacheKey.js compile > .build/packagelockhash
+  - script: mkdir -p .build && node build/azure-pipelines/common/computeNodeModulesCacheKey.js compile $VSCODE_ARCH > .build/packagelockhash
```

Comment:
We usually just bump the value in https://github.com/microsoft/vscode/blob/main/build/.cachesalt to nuke the cache

Reply:
All other stages are using the ARCH as part of the cache key. 
Doing the same here will help us in case we have to switch the Compile stage again.